### PR TITLE
Add Delivery Tracking for Push Notifications

### PR DESCRIPTION
**Pull request by cgwyllie**
### Description of Changes

See:

- #53 
- #55 
- #56 
- #57 

This PR pulls all changes into one place for final review, test, merge, & release on master.

Additional minor fixes:

- Background event sync race condition
- Delivery for content-available
- Support for badge increments (#59)

### Breaking Changes

-   None

### Release Checklist

Prepare:

-   [x] Detail any breaking changes. Breaking changes require a new major version number
-   [x] Check `pod lib lint` passes

Bump versions in:

-   [x] `Sources/Kumulos.swift`
-   [x] `KumulosSdkSwift.podspec`
-   [x] `KumulosSdkSwiftExtension.podspec`
-   [x] `Build target version for plist`
-   [x] `README.md`

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Run `pod trunk push` to publish to CocoaPods

### CODE REVIEW

File: Sources/Shared/KelValPersistenceHelper.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
+//
```
Comment (azlindsay) [inline Sources/Shared/KelValPersistenceHelper.swift:1]: Filename -> KeyVal

File: Sources/Kumulos+Analytics.swift [hunk 2]
```diff
@@ -8,71 +8,68 @@
...
 public extension Kumulos {
-
-    private static let userIdLock = DispatchSemaphore(value: 1)
-    internal static let USER_ID_KEY = "KumulosCurrentUserID"
-    
     internal static func trackEvent(eventType: KumulosEvent, properties: [String:Any]?, immediateFlush: Bool = false) {
         getInstance().analyticsHelper.trackEvent(eventType: eventType.rawValue, properties: properties, immediateFlush: immediateFlush)
     }
     
+    internal static func trackEvent(eventType: KumulosSharedEvent, properties: [String:Any]?, immediateFlush: Bool = false) {
```
Comment (azlindsay) [inline Sources/Kumulos+Analytics.swift:15]: Just checking this has been sync-merged with master - we did a patch recently when we removed default values for these helpers?

Comment (cgwyllie) [reply Sources/Kumulos+Analytics.swift:15]: Eh, good question.. they were only removed at some level, think inside AnalyticsHelper, maybe not this level?

Comment (cgwyllie) [reply Sources/Kumulos+Analytics.swift:15]: https://github.com/Kumulos/KumulosSdkSwift/pull/58/commits/24402223b12a56ba5c45c051a64f819d5d3e0399

File: Sources/Kumulos+Push.swift [hunk 3]
```diff
@@ -387,4 +393,17 @@ class PushHelper {
...
+    
+    fileprivate func setBadge(userInfo: [AnyHashable:Any]){
+        let badge: NSNumber? = KumulosHelper.getBadgeFromUserInfo(userInfo: userInfo)
+        if let newBadge = badge {
+            UIApplication.shared.applicationIconBadgeNumber = newBadge.intValue
+        }
+    }
+    
+    fileprivate func trackPushDelivery(userInfo: [AnyHashable : Any]){
+        let notification = KSPushNotification(userInfo: userInfo)
```
Comment (cgwyllie) [inline Sources/Kumulos+Push.swift:405]: See in ObjC this is guarded in case not one of our notifications / factory fails, same applicable here?

Comment (ftcvlad) [reply Sources/Kumulos+Push.swift:405]: k

**Pull request merged**